### PR TITLE
Feature/jax advantage estimation utility unit test 

### LINKED ISSUE #533: [TEST] Jax Advantage Estimation
### What do you want to test?
Jax gae components

### Outline of test structure
* Unit tests
* Test components and hooks

### Definition of done
Passing checks, cover all hooks, edge cases considered

### Mandatory checklist before making a PR
* [ ] The success criteria laid down in “Definition of done” are met.
* [ ] Test code is documented - docstrings for methods and classes, static types for arguments.
* [ ] Documentation is updated - README, CONTRIBUTING, or other documentation.

### PR DESCRIPTION
**Pull request by RuanJohn**
## What?
Unit tests for the general advantge estimation component.
## Why?
Expand testing in Mava redesign

closes #533

### CODE REVIEW

**Review by mmorris44 — CHANGES_REQUESTED** [review]
Thanks @RuanJohn! Tests themselves are great.
I've left a few comments about formatting and docstrings

File: tests/jax/components/training/advantage_estimation_test.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+import pytest
+
+from mava.components.jax.training.advantage_estimation import GAE, GAEConfig
+from mava.core_jax import SystemTrainer
+
+
+def different_reward_values() -> List[Tuple]:
+    """Create test case data with different reward scalings"""
+
+    return [
```
[the remark this answers is not in the document]
Comment (RuanJohn) [reply tests/jax/components/training/advantage_estimation_test.py:30]: Me too, but the linter seems to not allow for it.

File: tests/jax/components/training/advantage_estimation_test.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    """Abstract system trainer."""
+
+    def __init__(
+        self,
+    ) -> None:
+        """System trainer init"""
+
+        self.store = SimpleNamespace(gae_fn=None)
+
+        self._inputs: Any
```
[the remark this answers is not in the document]
Comment (RuanJohn) [reply tests/jax/components/training/advantage_estimation_test.py:119]: Removed the entire mocked trainer and inhereting from `Trainer` directly.

File: tests/jax/components/training/advantage_estimation_test.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+                    0.3,
+                    0.2,
+                    0.3,
+                ]
+            ),
+        )
+    ]
+
+
+class MockTrainer(SystemTrainer):
```
[the remark this answers is not in the document]
Comment (RuanJohn) [reply tests/jax/components/training/advantage_estimation_test.py:109]: Agreed. Done :muscle:

File: tests/jax/components/training/advantage_estimation_test.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    """Abstract system trainer."""
+
+    def __init__(
+        self,
+    ) -> None:
+        """System trainer init"""
+
+        self.store = SimpleNamespace(gae_fn=None)
```
[the remark this answers is not in the document]
Comment (RuanJohn) [reply tests/jax/components/training/advantage_estimation_test.py:117]: Sorted.

File: tests/jax/components/training/advantage_estimation_test.py [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+
+def test_gae_creation() -> None:
+    """Test whether gae function is successfully created"""
+
+    test_gae = GAE()
+    mock_trainer = MockTrainer()
+    test_gae.on_training_utility_fns(trainer=mock_trainer)
+
+    assert mock_trainer.store.gae_fn is not None
```
[the remark this answers is not in the document]
Comment (RuanJohn) [reply tests/jax/components/training/advantage_estimation_test.py:133]: Agreed. Done.

File: tests/jax/components/training/advantage_estimation_test.py [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+    def step(self) -> None:
+        """Trainer forward and backward passes."""
+        pass
+
+
+def test_gae_creation() -> None:
+    """Test whether gae function is successfully created"""
+
+    test_gae = GAE()
+    mock_trainer = MockTrainer()
```
[the remark this answers is not in the document]
Comment (RuanJohn) [reply tests/jax/components/training/advantage_estimation_test.py:130]: Agreed. Made `mock_trainer`, `gae_with_reward_clipping` and `gae_without_reward_clipping` fixtures to be used across all tests.

File: tests/jax/components/training/advantage_estimation_test.py [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+    """Test whether reward clipping in gae_advantages is working
+
+    Done by verifying whether it is returning similar advantage
+    and target values when given rewards with different scales.
+    """
+
+    test_gae = GAE(config=GAEConfig(max_abs_reward=1.0))
+    mock_trainer = MockTrainer()
+    test_gae.on_training_utility_fns(trainer=mock_trainer)
+
```
[the remark this answers is not in the document]
Comment (RuanJohn) [reply tests/jax/components/training/advantage_estimation_test.py:154]: Decreased whitespace where it seemed sensible to do so :call_me_hand:

File: tests/jax/components/training/advantage_estimation_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmorris44) [reply tests/jax/components/training/advantage_estimation_test.py:30]: Sadness

**Review by mmorris44 — APPROVED** [review]
Amazing PR and very neat code! Thanks @RuanJohn :)

**Review by siddarthsingh1 — APPROVED** [review]
